### PR TITLE
Add lux.js example

### PR DESCRIPTION
**Pull request by ifandelse**
@voronianski Thank you so much for creating this project! I've added an example implementation using [lux.js](https://github.com/LeanKit-Labs/lux.js). Please let me know if you need me to change anything, or if you have any questions. Thanks!

### CODE REVIEW

File: lux/js/stores/CartStore.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                this.setState( newState );
+            }
+        },
+        cartCheckout: function() {
+            this.replaceState( { products: {} } );
+        },
+        successCheckout: function( products ) {
+            // this can be used to redirect to success page, etc.
+            console.log( 'YOU BOUGHT:' );
+            if ( typeof console.table === "function" ) {
```
Comment (voronianski) [inline lux/js/stores/CartStore.js]: @ifandelse also remove  (see [contributing guidelines](https://github.com/voronianski/flux-comparison/blob/master/CONTRIBUTING.md#there-are-not-too-much-of-rules-to-follow)).

File: lux/js/app.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+require( 'es5-shim' );
+require( 'es5-shim/es5-sham' );
+require( 'babel/polyfill' );
```
Comment (voronianski) [inline lux/js/app.js]: @ifandelse please could you remove all spaces from your function call statements, see [contributing guidelines](https://github.com/voronianski/flux-comparison/blob/master/CONTRIBUTING.md#there-are-not-too-much-of-rules-to-follow).

File: lux/js/stores/ProductStore.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var lux = require( 'lux.js' );
+var ProductStore = new lux.Store( {
+
+    state: {
+        products: []
+    },
+
+    namespace: "products",
```
Comment (voronianski) [inline lux/js/stores/ProductStore.js]: @ifandelse single quotes please

File: lux/js/stores/ProductStore.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    handlers: {
+        receiveProducts: function( products ) {
+            this.setState( { products: products } );
+        },
+        addToCart: function( product ) {
```
Comment (voronianski) [inline lux/js/stores/ProductStore.js]: @ifandelse remove spaces

File: lux/js/components/ProductsContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return getAllProducts();
+    },
+
+    render: function () {
+        var nodes = this.state.products.map(function (product) {
+            return <ProductItemContainer key={product.id} product={product} />;
+        });
+
+        return (
+            <ProductsList title="Flux Shop Demo (Facebook)">
```
Comment (voronianski) [inline lux/js/components/ProductsContainer.jsx]: @ifandelse title should be `"Flux Shop Demo (Lux)"`.

**Comment by voronianski** [comment]
@ifandelse thanks a lot for this PR!

Could you please fix the style issues and rebase with master in order to be merged like charm :)

**Comment by ifandelse** [comment]
@voronianski apologies - we have some aggressive esformatter settings which I completely forgot to disable. Happy to update and rebase. Traveling today, so it may be this afternoon before I can finish.

File: lux/js/stores/CartStore.js (hunk 1 above)
Comment (ifandelse) [inline lux/js/stores/CartStore.js]: @voronianski just to make sure I'm clear - remove the console.table? Or was this referencing the spaces in the arg list and inside the if?

Comment (voronianski) [inline lux/js/stores/CartStore.js]: @ifandelse sorry I mean spaces, table is pretty cool :)

**Comment by ifandelse** [comment]
@voronianski OK - I rebased - please let me know if you need anything else changed. Thanks! :smile:

**Comment by voronianski** [comment]
@ifandelse everything's fine, just add `lux/build/*.js` to `.gitignore` please (https://github.com/voronianski/flux-comparison/blob/master/.gitignore#L12). Thanks! :smiley:

**Comment by ifandelse** [comment]
@voronianski no problem - just updated the PR, thanks!

**Comment by voronianski** [comment]
@ifandelse :+1: merged

**Comment by ifandelse** [comment]
@voronianski The lux.js example doesn't work because of [this line being removed recently](https://github.com/voronianski/flux-comparison/commit/8728017bd791863f9d190eacb27c583327245678#diff-e0b14e101392b2b6f44be480a9a0b798L9). I'm sure it was just a simple mistake, no worries. I will PR here shortly and add it back in.

**Pull request merged**